subcommand/inject-connect/command.go

```diff
@@ -145,8 +155,18 @@ func (c *Command) init() {
 	c.flagSet.StringVar(&c.flagLifecycleSidecarMemoryLimit, "lifecycle-sidecar-memory-limit", "50Mi", "Lifecycle sidecar memory limit.")
 
 	c.http = &flags.HTTPFlags{}
+	c.k8s = &flags.K8SFlags{}
+
 	flags.Merge(c.flagSet, c.http.Flags())
+	flags.Merge(c.flagSet, c.k8s.Flags())
 	c.help = flags.Usage(help, c.flagSet)
+
+	// wait on an interrupt for exit, be sure to init it before running
+	// the controller so that we dont receive an interrupt before its ready
+	if c.sigCh == nil {
```

Comment:
We need to be using this to trigger server, cert-watcher and health check controller shutdown.